src/RootStore/UserStore.ts

```diff
@@ -190,8 +195,14 @@ export default class UserStore {
   get routes(): RoutePermission[] {
     if (!this.userAppMetadata?.routes) return [];
     const routePermissions = entries(this.userAppMetadata?.routes);
+    const routes = routePermissions.map((r) => {
+      const [fullRoute, permission] = r;
```

Comment:
If you'd like, you can do the array destructuring right in the function arguments and avoid needing to make an intermediate variable.

```suggestion
    const routes = routePermissions.map(([fullRoute, permission]) => {
```